snapcraft/internal/project_loader/_parts_config.py

```diff
@@ -252,23 +249,26 @@ def build_env_for_part(self, part, root_part=True) -> List[str]:
 
         env = []  # type: List[str]
         stagedir = self._project.stage_dir
-        is_host_compat = self._project.is_host_compatible_with_base(self._base)
 
         if root_part:
             # this has to come before any {}/usr/bin
             env += part.env(part.plugin.installdir)
             env += runtime_env(part.plugin.installdir, self._project.arch_triplet)
             env += runtime_env(stagedir, self._project.arch_triplet)
             env += build_env(
-                part.plugin.installdir, self._snap_name, self._project.arch_triplet
+                part.plugin.installdir,
+                self._project.info.name,
+                self._project.arch_triplet,
```

Comment:
There are a lot of changes in this PR that don't actually seem to relate to the issue being fixed, it makes the review more difficult.

Reply:
I changed base as it was now technically incorrect and thought I'd remove the other two props

Reply:
I did the code count @kyrofa, and I believe you are probably exagerating, it is 7 lines of code that really changed do to using `self._project.info.name` and `self._project.info.confinement`.

Not defaulting to `core` in the base is somewhat of a requirement as it is not a base.